test/e2e/ipsec.go

```diff
@@ -0,0 +1,160 @@
+				"bash", "-c", "sudo pkill pluto")
+			framework.ExpectNoError(err, "killing pluto process failed")
+			time.Sleep(10 * time.Second)
+
+			for _, nodeInfo := range nodesIPinfo {
+				metricsOutput := getIPSecMetricsFromNode(f, nodeInfo.name)
+
+				ginkgo.By(fmt.Sprintf("Verify ipsec tunnel metrics reflecting down from node %s. ", nodeInfo.name))
+				gomega.Expect(metricsOutput).Should(gomega.ContainSubstring(ipsecTunnelMetrics[1]))
+			}
+
+			// pluto process will be back automatically.
+			ginkgo.By(fmt.Sprintf("Wait pluto process back on node %s", node1.name))
+			gomega.Eventually(func() bool {
+				output, _ := e2ekubectl.RunKubectl(ovnKubernetesNamespace, "exec", ipsecPod.Items[0].Name, "--container", ipsecContainer, "--",
+					"bash", "-c", "sudo ps -ef |grep pluto")
+				return strings.Contains(output, " /usr/libexec/ipsec/pluto --leak-detective --config /etc/ipsec.conf")
+			}, 120*time.Second, 10*time.Second).Should(gomega.BeTrue())
+
+			ginkgo.By("Verify ipsec metrics reflecting the up tunnel for all nodes")
+			//Wait for the metrics reflecting the tunnel up for all the nodes
+			gomega.Eventually(func() bool {
+				for _, nodeInfo := range nodesIPinfo {
+					metricsOutput := getIPSecMetricsFromNode(f, nodeInfo.name)
+
+					ginkgo.By(fmt.Sprintf("Verify ipsec tunnel metrics reflecting up again from node %s.", nodeInfo.name))
+					if !strings.Contains(metricsOutput, ipsecTunnelMetrics[0]) {
+						return false
+					}
+				}
+				return true
+			}, 240*time.Second, 30*time.Second).Should(gomega.BeTrue())
+		})
+
+	})
+
+})
+
+// getIPSecMetricsFromNode retrieves IPSec metrics from the specified node
+func getIPSecMetricsFromNode(f *framework.Framework, nodeName string) string {
+	ovnKubernetesNamespace := deploymentconfig.Get().OVNKubernetesNamespace()
+	containerName := "ovnkube-node"
+	if isInterconnectEnabled() {
+		containerName = "ovnkube-controller"
+	}
+
+	ovnKubeNodePods, err := f.ClientSet.CoreV1().Pods(ovnKubernetesNamespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "name=ovnkube-node",
+		FieldSelector: "spec.nodeName=" + nodeName,
+	})
+	if err != nil {
+		framework.Failf("could not get ovnkube-node pods: %v", err)
+	}
+	if len(ovnKubeNodePods.Items) == 0 {
+		framework.Failf("no ovnkube-node pods found on node %s", nodeName)
+	}
+	ovnPodName := ovnKubeNodePods.Items[0].Name
+
+	framework.Logf("Check ipsec metrics on ovnkube pod %s", ovnPodName)
+	url := fmt.Sprintf("http://%s/metrics", net.JoinHostPort(nodeName, "9410"))
+	metricsOutput, err := e2ekubectl.RunKubectl(ovnKubernetesNamespace, "exec", ovnPodName, "--container", containerName, "--",
+		"bash", "-c", fmt.Sprintf("curl --connect-timeout 2 %s | grep ipsec", url))
+	framework.ExpectNoError(err, "Retrieving metrics failed")
+
+	return metricsOutput
+}
+
+func collectNodesAddresses(nodes *v1.NodeList, addressType v1.NodeAddressType) []nodeInfo {
```

Comment:
do we need a separate function for this ? could be done it in `BeforeEach` itself.
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/test/e2e/egress_firewall.go#L94-L108